File: client/src/services/category.service.js
import { axiosClient } from '~/api';

const branch_api = "/categories";

export const categoryService = {
    getAllCategories() {
        return axiosClient.get(`${branch_api}`);
    },
    getAllCategoriesByShopId(shopId) {
        return axiosClient.get(`${branch_api}/shopId/${shopId}`);
    },
    getCategoryBySlug(slug) {
        return axiosClient.get(`${branch_api}/slug/${slug}`);
    },
};

File: backend/src/main/java/gt/electronic/ecommerce/services/CategoryService.java
package gt.electronic.ecommerce.services;

import gt.electronic.ecommerce.dto.request.CategoryCreationDTO;
import gt.electronic.ecommerce.dto.response.CategoryResponseDTO;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

/**
 * @author quang huy
 * @created 11/09/2025 - 10:01 AM
 */
public interface CategoryService {
  List<CategoryResponseDTO> getAllCategories(String keyword);

  List<CategoryResponseDTO> getAllCategoriesByShop(Long shopId);

  CategoryResponseDTO getCategoryById(Integer id);

  CategoryResponseDTO getCategoryBySlug(String slug);

  CategoryResponseDTO createCategory(CategoryCreationDTO creationDTO, MultipartFile imageFile);

  CategoryResponseDTO updateCategory(Integer id, CategoryCreationDTO creationDTO, MultipartFile imageFile);

  CategoryResponseDTO deleteCategoryById(Integer id);
}
